lib/toolbox/bit_buffer.h

```diff
@@ -10,134 +16,146 @@ extern "C" {
+ * @param[in]     other      pointer to a BitBuffer instance to copy from
+ * @param[in]     end_index  index to end copying source data at
  */
 void bit_buffer_copy_left(BitBuffer* buf, const BitBuffer* other, size_t end_index);
 
-/**
- * Copy a byte array to a BitBuffer instance, replacing all of the original data.
- * The destination capacity must be no less than the source data size.
+/** Copy a byte array to a BitBuffer instance, replacing all of the original
+ * data.
+ *
+ * @warning       The destination capacity must be no less than the source data
+ *                size.
  *
- * @param [in,out] buf pointer to a BitBuffer instance to copy into
- * @param [in] data pointer to the byte array to be copied
- * @param [in] size_bytes size of the data to be copied, in bytes
+ * @param[in,out] buf         pointer to a BitBuffer instance to copy into
+ * @param[in]     data        pointer to the byte array to be copied
+ * @param[in]     size_bytes  size of the data to be copied, in bytes
  */
 void bit_buffer_copy_bytes(BitBuffer* buf, const uint8_t* data, size_t size_bytes);
 
-/**
- * Copy a byte array to a BitBuffer instance, replacing all of the original data.
- * The destination capacity must be no less than the source data size.
+/** Copy a byte array to a BitBuffer instance, replacing all of the original
+ * data.
+ *
+ * @warning       The destination capacity must be no less than the source data
+ *                size.
  *
- * @param [in,out] buf pointer to a BitBuffer instance to copy into
- * @param [in] data pointer to the byte array to be copied
- * @param [in] size_bits size of the data to be copied, in bits
+ * @param[in,out] buf        pointer to a BitBuffer instance to copy into
+ * @param[in]     data       pointer to the byte array to be copied
+ * @param[in]     size_bits  size of the data to be copied, in bits
  */
 void bit_buffer_copy_bits(BitBuffer* buf, const uint8_t* data, size_t size_bits);
 
-/**
- * Copy a byte with parity array to a BitBuffer instance, replacing all of the original data.
- * The destination capacity must be no less than the source data size.
+/** Copy a byte with parity array to a BitBuffer instance, replacing all of the
+ * original data.
  *
- * @param [in,out] buf pointer to a BitBuffer instance to copy into
- * @param [in] data pointer to the byte array to be copied
- * @param [in] size_bitss size of the data to be copied, in bits
+ * @warning       The destination capacity must be no less than the source data
+ *                size.
+ *
+ * @param[in,out] buf        pointer to a BitBuffer instance to copy into
+ * @param[in]     data       pointer to the byte array to be copied
+ * @param[in]     size_bits  size of the data to be copied, in bits
+ * @note          Parity bits are placed starting with the least significant bit
+ *                of each byte and moving up.
+ * @note          Example: PDDDDDDD DPDDDDDD DDPD...
```

Comment:
The parity bit follows the 8 data bits and is packed into an uint8_t array.
DDDDDDDDP     DDDDDDDDP     DDDDDDDDP ... -> DDDDDDDD     PDDDDDDD     DPDDDDDD     DDP ....
MSB -> LSB

Reply:
Please fix the description

Reply:
As mentioned above, this is MSB